cli/src/merge_tools/external.rs

```diff
@@ -427,19 +431,38 @@ pub fn generate_diff(
         .map_err(ExternalToolError::SetUpDir)?;
     set_readonly_recursively(diff_wc.right_working_copy_path())
         .map_err(ExternalToolError::SetUpDir)?;
-    invoke_external_diff(ui, writer, tool, &diff_wc.to_command_variables())
+    invoke_external_diff(
+        ui,
+        writer,
+        tool,
+        diff_wc.temp_dir(),
+        &diff_wc.to_command_variables(),
+    )
 }
 
 /// Invokes the specified `tool` directing its output into `writer`.
 pub fn invoke_external_diff(
     ui: &Ui,
     writer: &mut dyn Write,
     tool: &ExternalMergeTool,
+    diff_dir: &Path,
     patterns: &HashMap<&str, &str>,
 ) -> Result<(), DiffGenerateError> {
     // TODO: Somehow propagate --color to the external command?
     let mut cmd = Command::new(&tool.program);
-    cmd.args(interpolate_variables(&tool.diff_args, patterns));
+    if tool.diff_do_chdir {
+        cmd.args(interpolate_variables(
+            &tool.diff_args,
+            &maplit::hashmap! {
+                "left" => "left",
+                "right" => "right",
+            },
+        ));
+        cmd.current_dir(diff_dir);
+    } else {
+        cmd.args(interpolate_variables(&tool.diff_args, patterns));
+    }
+
     tracing::info!(?cmd, "Invoking the external diff generator:");
```

Comment:
I'm curious, if you run `jj diff --debug --tool something`, will this show the directory the program is run from? If not, we should show it manually, so that the user can reproduce the invocation and find the files if they need to debug the diff tool.

Reply:
It does:

```
2025-04-03T20:10:33.683720Z  INFO run_command:cmd_diff: jj_cli::merge_tools::external: Invoking the external diff generator: cmd=cd "/var/folders/1p/_w9l06b48xj8h8001s7n_mkh0000gn/T/jj-diff-caPuNI" && "something" "left" "right"
```

but the temporary directory is deleted before jj exits.